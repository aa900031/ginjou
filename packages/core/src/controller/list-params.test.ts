import { describe, expect, expectTypeOf, it, vi } from 'vitest'
import type { ListParams } from '.'
import { createListParams } from '.'

describe('test List Params', () => {
	it('should default value', () => {
		const params = createListParams()
		expect(params.getValue()).toMatchSnapshot()
	})

	it('should emit "change" after update value', () => {
		const params = createListParams()
		const handler = vi.fn().mockImplementation((event: { value: ListParams }) => {
			expect(event.value.pagination.current).toBe(2)
			expectTypeOf(event.value).toEqualTypeOf<ListParams>()
		})
		params.on('change', handler)
		params.setPage(2)

		expect(handler).toBeCalledTimes(1)
	})
})
