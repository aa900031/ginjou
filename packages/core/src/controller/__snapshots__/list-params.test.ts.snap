// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test List Params > should default value 1`] = `
{
  "filters": null,
  "pagination": {
    "current": 1,
    "perPage": 10,
  },
  "sorters": [
    {
      "field": "id",
      "order": "asc",
    },
  ],
}
`;
